### PR TITLE
#1: OdpDraft -> OdpVersion

### PR DESCRIPTION
**Pull request by sorja**
- Remove OdpDraftRepository
- Introduce OdpVersionRepository
- Use persistDraft under service to start transaction

Part of #806

### CODE REVIEW

File: server/repository/OdpVersion/update.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { BaseProtocol, DB } from '@server/db'
+
+export const update = async (options: { draft: any; draftId: any }, client: BaseProtocol = DB) => {
```
Comment (minotogna) [inline server/repository/OdpVersion/update.ts:3]: we should start avoiding any.

File: server/service/odp/persistDraft.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  await OdpClassRepository.addClassData({ odpVersionId, odp: draft }, client)
+  await OdpRepository.updateDraftId({ odpId }, client)
+}
+
+const updateOrInsertDraft = async (
+  options: { user: any; odpId: any; countryIso: any; draft: any },
+  client: BaseProtocol = DB
+) => {
+  const { user, odpId, countryIso, draft } = options
+  const draftId = await OdpRepository.getDraftId(client, odpId)
```
Comment (minotogna) [inline server/service/odp/persistDraft.ts:32]: client must be always the second argument. the first always an object

Comment (sorja) [reply server/service/odp/persistDraft.ts:33]: handled in separate pr

File: server/api/odp/createDraft.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ export const OdpCreateDraft = {
       async (req: Request, res: Response) => {
         try {
           const { countryIso }: any = req.query
-          const result = await OdpService.saveDraft({ countryIso, user: req.user as any, draft: req.body })
+          const result = await OdpService.persistDraft({ countryIso, user: req.user as any, draft: req.body })
```
Comment (minotogna) [inline server/api/odp/createDraft.ts:15]: do we have user type?

File: server/service/odp/persistDraft.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+const updateOrInsertDraft = async (
+  options: { user: User; odpId: number; countryIso: CountryIso; draft: ODP },
+  client: BaseProtocol = DB
+) => {
+  const { user, odpId, countryIso, draft } = options
+  const draftId = await OdpRepository.getDraftId(client, odpId)
+
+  if (draftId) {
+    updateDraft({ draft })
```
Comment (minotogna) [inline server/service/odp/persistDraft.ts:36]: client must be passed to updateDraft

**Review by minotogna — APPROVED** [review]

**Pull request merged**